tests/e2e/vsphere_volume_expansion.go

```diff
@@ -199,8 +238,442 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+		defer cancel()
+
+		ginkgo.By("Invoking Test for Volume Expansion")
+
+		volHandle, pvclaim, pv, storageclass := createStorageClassAndPVC(f, client, namespace)
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		pod := createPODandVerifyVolumeMount(f, client, namespace, pvclaim, volHandle)
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		}()
+
+		increaseVolumeMultipleTimes(f, client, namespace, volHandle, pvclaim, pod)
+	})
+
 })
 
+//increaseVolumeMultipleTimes this method increases the same volume multiple times and verifies PVC and Filesystem size
+func increaseVolumeMultipleTimes(f *framework.Framework, client clientset.Interface, namespace string, volHandle string, pvclaim *v1.PersistentVolumeClaim, pod *v1.Pod) {
+	// Modify PVC spec to trigger volume expansion
+	ginkgo.By("Expanding pvc 10 times")
+	currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+	newSize := currentPvcSize.DeepCopy()
+	for i := 0; i < 10; i++ {
+		newSize.Add(resource.MustParse("1Gi"))
+		ginkgo.By(fmt.Sprintf("Expanding pvc to new size: %v", newSize))
+		pvclaim, err := expandPVCSize(pvclaim, newSize, client)
+		framework.ExpectNoError(err, "While updating pvc for more size")
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		pvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) != 0 {
+			framework.Failf("error updating pvc %q to size %v", pvclaim.Name, newSize)
+		}
+	}
+
+	ginkgo.By("Waiting for controller resize to finish")
+	framework.Logf("PVC name : " + pvclaim.Name)
+	pv := getPvFromClaim(client, namespace, pvclaim.Name)
+	pvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+	err := waitForPvResize(pv, client, pvcSize, totalResizeWaitPeriod)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Checking for conditions on pvc")
+	framework.Logf("PVC Name :", pvclaim.Name)
+	pvclaim, err = waitForPVCToReachFileSystemResizePendingCondition(client, namespace, pvclaim.Name, totalResizeWaitPeriod)
```

Comment:
As this is online volume resize, have you checked for a possibility where the PVC might have resized successfully by this point and the test might fail here assuming the ResizePending condition is not there?

Reply:
Did you run the tests at least 2 or 3 times to check if they are consistently passing? If yes, then we are good with this code

Reply:
yes i have run it thrice locally and once in jenkin